executor/union_scan.go

```diff
@@ -139,44 +139,59 @@ func (us *UnionScanExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	defer us.memBuf.RUnlock()
 	req.GrowAndReset(us.maxChunkSize)
 	mutableRow := chunk.MutRowFromTypes(retTypes(us))
+
+	var done bool
+	var err error
+	for !done {
+		done, err = us.nextOnce(ctx, mutableRow, req)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// nextOnce handles a batch of chunk rows, returns whether whole work is done.
+func (us *UnionScanExec) nextOnce(ctx context.Context, mutableRow chunk.MutRow, req *chunk.Chunk) (bool, error) {
 	for i, batchSize := 0, req.Capacity(); i < batchSize; i++ {
```

Comment:
Will the change more simple if we do this:

```
for batchSize := req.Capacity(); req.NumRows() < batchSize; {
```

Reply:
Done